pkg/store/s3.go

```diff
@@ -110,12 +112,22 @@ func DownloadFromS3(cred S3Credentials, src string, dest string) (string, error)
 		return "", trace.Wrap(err)
 	}
 
+	s3Object, err := client.GetObject(loc.Bucket, loc.Path)
```

Comment:
I think, you can replace this code and io.Copy below with http://docs.minio.io/docs/golang-client-api-reference#FGetObject.